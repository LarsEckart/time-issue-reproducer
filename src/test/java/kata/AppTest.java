package kata;

import com.spun.util.DateDifference;
import com.spun.util.DateUtils;
import com.spun.util.parser.TemplateDate;
import org.approvaltests.Approvals;
import org.approvaltests.core.Options;
import org.approvaltests.reporters.Junit5Reporter;
import org.approvaltests.utils.WithTimeZone;
import org.junit.jupiter.api.DisplayNameGeneration;
import org.junit.jupiter.api.DisplayNameGenerator;
import org.junit.jupiter.api.RepeatedTest;
import org.junit.jupiter.api.Test;

import java.sql.Timestamp;
import java.text.DateFormat;

import static org.assertj.core.api.Assertions.assertThat;

@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
class AppTest {

    @RepeatedTest(10000)
    public void testFebruaryAndDaylightSavingsTime()
    {
        try (WithTimeZone tz = new WithTimeZone("PST"))
        {
            StringBuilder buffer = new StringBuilder();
            DateFormat f = TemplateDate.FORMATS.DATE_SHORT;
            for (int i = 1; i <= 28; i++)
            {
                Timestamp a = DateUtils.parse("2010/02/0" + i);
                Timestamp b = DateUtils.parse("2010/03/0" + i);
                DateDifference dif = new DateDifference(a, b);
                String standardTimeText = dif.getStandardTimeText(2, "days", "seconds", null, null);
                buffer.append(String.format("%s, %s => %s (%s, %s) \n", f.format(a), f.format(b), standardTimeText, a.getTime(), b.getTime()));
            }
            Approvals.verify(buffer.toString(), new Options(new Junit5Reporter()));
        }
    }
}
