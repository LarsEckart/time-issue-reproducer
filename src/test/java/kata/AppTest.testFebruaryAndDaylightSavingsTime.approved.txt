02/01/2010, 03/01/2010 => 28 Days, 0 Hours (1265011200000, 1267430400000) 
02/02/2010, 03/02/2010 => 28 Days, 0 Hours (1265097600000, 1267516800000) 
02/03/2010, 03/03/2010 => 28 Days, 0 Hours (1265184000000, 1267603200000) 
02/04/2010, 03/04/2010 => 28 Days, 0 Hours (1265270400000, 1267689600000) 
02/05/2010, 03/05/2010 => 28 Days, 0 Hours (1265356800000, 1267776000000) 
02/06/2010, 03/06/2010 => 28 Days, 0 Hours (1265443200000, 1267862400000) 
02/07/2010, 03/07/2010 => 28 Days, 0 Hours (1265529600000, 1267948800000) 
02/08/2010, 03/08/2010 => 28 Days, 0 Hours (1265616000000, 1268035200000) 
02/09/2010, 03/09/2010 => 28 Days, 0 Hours (1265702400000, 1268121600000) 
02/10/2010, 03/10/2010 => 28 Days, 0 Hours (1265788800000, 1268208000000) 
02/11/2010, 03/11/2010 => 28 Days, 0 Hours (1265875200000, 1268294400000) 
02/12/2010, 03/12/2010 => 28 Days, 0 Hours (1265961600000, 1268380800000) 
02/13/2010, 03/13/2010 => 28 Days, 0 Hours (1266048000000, 1268467200000) 
02/14/2010, 03/14/2010 => 28 Days, 0 Hours (1266134400000, 1268553600000) 
02/15/2010, 03/15/2010 => 27 Days, 23 Hours (1266220800000, 1268636400000) 
02/16/2010, 03/16/2010 => 27 Days, 23 Hours (1266307200000, 1268722800000) 
02/17/2010, 03/17/2010 => 27 Days, 23 Hours (1266393600000, 1268809200000) 
02/18/2010, 03/18/2010 => 27 Days, 23 Hours (1266480000000, 1268895600000) 
02/19/2010, 03/19/2010 => 27 Days, 23 Hours (1266566400000, 1268982000000) 
02/20/2010, 03/20/2010 => 27 Days, 23 Hours (1266652800000, 1269068400000) 
02/21/2010, 03/21/2010 => 27 Days, 23 Hours (1266739200000, 1269154800000) 
02/22/2010, 03/22/2010 => 27 Days, 23 Hours (1266825600000, 1269241200000) 
02/23/2010, 03/23/2010 => 27 Days, 23 Hours (1266912000000, 1269327600000) 
02/24/2010, 03/24/2010 => 27 Days, 23 Hours (1266998400000, 1269414000000) 
02/25/2010, 03/25/2010 => 27 Days, 23 Hours (1267084800000, 1269500400000) 
02/26/2010, 03/26/2010 => 27 Days, 23 Hours (1267171200000, 1269586800000) 
02/27/2010, 03/27/2010 => 27 Days, 23 Hours (1267257600000, 1269673200000) 
02/28/2010, 03/28/2010 => 27 Days, 23 Hours (1267344000000, 1269759600000) 
